core/services/directrequest/delegate.go

```diff
@@ -284,11 +295,16 @@ func (l *listener) handleOracleRequest(request *oracle_wrapper.OracleOracleReque
 }
 
 // Cancels runs that haven't been started yet, with the given request ID
-func (l *listener) handleCancelOracleRequest(request *oracle_wrapper.OracleCancelOracleRequest) {
+func (l *listener) handleCancelOracleRequest(request *oracle_wrapper.OracleCancelOracleRequest, lb log.Broadcast) {
 	runCloserChannelIf, loaded := l.runs.LoadAndDelete(formatRequestId(request.RequestId))
 	if loaded {
 		close(runCloserChannelIf.(chan struct{}))
 	}
+	ctx, cancel := postgres.DefaultQueryCtx()
```

Comment:
hmm, can this race with Execute run?

Reply:
yeah I think it can, it did before this change as well, but you either get the cancel request in time to stop the run or you don't either way we mark the log consumed